core/src/main/kotlin/fr/sncf/osrd/conflicts/SpacingResourceGenerator.kt

```diff
@@ -2,302 +2,350 @@ package fr.sncf.osrd.conflicts
+                }
+            ongoingRequirementData.updateMinRequirementOffset(requiredOffset)
         }
-        return incrementalPath.getBlockEndZone(signal.minBlockPathIndex + 1) - 1
     }
 
-    // create requirements for zones occupied by the train at the beginning of the simulation
-    private fun setupInitialRequirements() {
-        if (lastEmittedZone != -1) return
-
-        // look for the zone which contains the train as it starts.
-        // when the simulation starts, the train occupies a single point on the tracks, as if it
-        // were contained
-        // inside a magic portal until the train entirely moved out of it
-        var startZone = -1
-        val startingPoint = incrementalPath.travelledPathBegin
-        for (i in 0 until incrementalPath.zonePathCount) {
-            val zonePathStartOffset = incrementalPath.getZonePathStartOffset(i)
-            val zonePathEndOffset = incrementalPath.getZonePathEndOffset(i)
-            if (startingPoint >= zonePathStartOffset && startingPoint < zonePathEndOffset) {
-                startZone = i
-                break
+    /** Returns the offset at which the ETCS "signal" can slow the train down. */
+    private fun getETCSFirstRequiredOffset(
+        simulationData: SimulationCacheData,
+        signalData: PendingSignalData,
+    ): Offset<TrainPath> {
+        val signal = signalData.signal
+        var isRouteDelimiter = true
+        try {
+            isRouteDelimiter = loadedSignalInfra.getSettings(signal).getFlag("Nf")
+        } catch (e: Throwable) {
+            logger.warn {
+                "Unable to determine if " +
+                    "signal ${rawInfra.getLogicalSignalName(signal)} is a route delimiter or not: $e"
             }
         }
-        assert(startZone != -1)
-
-        // We need to add a requirement for each zone between the start
-        // and the first required zone
-        val firstSignal = pendingSignals.firstOrNull()
-        val firstProtectedZone =
-            if (firstSignal != null) {
-                getSignalFirstProtectedZone(firstSignal)
-            } else {
-                startZone + 1
-            }
-
-        for (i in startZone until firstProtectedZone) {
-            // The simulation start time is 0.
-            // The zones are not all reached right at the start, but
-            // because they're part of the block the train starts on,
-            // they can be considered as "required" from the beginning.
-            addZonePendingRequirement(i, 0.0)
-            lastEmittedZone = i
+        val envelope = callbacks!!.getRawEnvelopeIfSingle()
+        // TODO: stop using a single envelope that's unavailable in STDCM and maybe move to a
+        //   dedicated EnvelopeTimeInterpolate.getIntersection().
+        //   Then probably protect its failure by returning
+        //   SignalRequirementsCreationStatus.NOT_SEEN_IN_AVAILABLE_SIM.
```

Comment:
This part of the comment is outdated, I'd go with something along:
```suggestion
        //   Then probably protect its failure like when simulation is insufficient.
```

Reply:
Fixed in https://github.com/OpenRailAssociation/osrd/pull/13873/commits/93cdf8f7f6c7b32ed77d482c11a920559094b4e0

I didn't use the exact same phrasing though